sdk/typespec/typespec_client_core/src/http/pager.rs

```diff
@@ -6,7 +6,33 @@ use std::{future::Future, pin::Pin};
 use futures::{stream::unfold, Stream};
 use typespec::Error;
 
-use crate::http::Response;
+use crate::http::{headers::HeaderName, Response};
+
+pub enum PagerResult<T, C> {
+    Continue {
+        response: Response<T>,
+        continuation: C,
+    },
+    Finish {
```

Comment:
Nit (but I could pick it up in another PR): most languages use "complete" for something like this.